TDD: feature/create-report

turtlelyte opened the pull request:
# Summary

### Files Added:
- docs/features/create_report.md (user story, MVC specifications, etc.)
- features/create_report.feature (Gherkin feature file, includes 3 scenarios)
- features/step_definitions/create_report_steps.rb (Ruby step definitions)
- features/screenshots/create_report_1.png (Cucumber test output)

### Tests:
- All tests fail as expected. Specific test scenarios listed in feature file

### Other:
- Added a new boolean column `admin` to the `users` table with default value being `false`.

StevenLi-phoenix commented on db/migrate/20251121005957_add_admin_to_users.rb:
why....
please use exist system
we have users.role

StevenLi-phoenix commented on spec/factories/users.rb:
please use exist system
we have users.role

StevenLi-phoenix commented:
Also remember to include your BDD features tests in ./cucumber.yaml once complete.

StevenLi-phoenix reviewed (changes requested):
Your cucumber test is failing:
      Ambiguous match of "I press "Search"":
      
      features/step_definitions/create_report_steps.rb:26:in `"I press {string}"'
      features/step_definitions/user_login_authentication_steps.rb:41:in `"I press {string}"'

other then that the code LGTM

turtlelyte commented:
I'm going to make the tests pass before before pushing to the main branch.

turtlelyte commented:
### Summary
- Implemented minimal model, views, and controllers to allow Cucumber and RSpec tests to pass.
- Changed implementation logic from creating new reports from user profile to having a separate view for creating new reports.
- Fixed ambiguous matches of step definitions.

### Notes/To-do:
- Admin and user authentication related tests are disabled for now as login view isn't implemented yet (correct me if I'm wrong).

mAE7777 commented on app/models/report.rb:
Quick thought - what happens if someone tries to report themselves? I think the Match model has something similar (cannot_match_self) if you want to take a look
